### PR TITLE
Improve windows support

### LINKED ISSUE #23: USB HID interface descriptor does not have an interface string
https://github.com/probe-rs/hs-probe-firmware/blob/master/firmware/src/usb/dap_v1.rs#L86

I don't fully understand why the `interface()` method here doesn't take a name, but the `interface_alt()` method we use on the v2 interface does; in any event CMSIS-DAP says we should have "CMSIS-DAP" in the interface string on composite devices, so we should probably add this somehow.

### LINKED ISSUE #25: Zadig needed to install drivers on Win10
I'm on Windows 10 build 19042.1165

After connecting the HS-Probe, Zadig was needed to install drivers for the CMSIS-DAP interfaces. Zadig showed `NONE` as the currently installed driver.
After installing WinUSB drivers, I was able to update the firmware to latest master with dfu-util and connect to a target for flashing/debugging etc.

The USB-VCOM doesn't show up in the COM-Port list, the device manager lists the device status as

```
This device cannot start. (Code 10)

A device which does not exist was specified.
```

I will try on other devices as well. Let me know if I can gather more information somehow

### PR DESCRIPTION
**Pull request by Tiwalun**
- Add a interface description string to the CMSIS-DAP v1 Interface and DFU interface
- Don't use fixed endpoints for the CMSIS-DAP v1 and v2 interface, so that they can be reordered.
- Use *interface association descriptors*, so that Windows works properly with the USB serial port.
- Setup [Microsoft OS 2.0 Descriptors](https://docs.microsoft.com/en-us/windows-hardware/drivers/usbcon/microsoft-os-2-0-descriptors-specification)
  With these descriptors, Windows will automatically use the WinUSB driver, without needing Zadig.

Fix #23 and fix #25 .

### CODE REVIEW

File: firmware/src/usb/winusb.rs [hunk 1]
```diff
@@ -1,80 +1,435 @@
...
-                        }
-                        _ => {
-                            xfer.reject().ok();
-                        }
-                    }
-                }
-                _ => {
-                    xfer.reject().ok();
-                }
+        // Check if the ve
```
Comment (Dirbaio) [inline firmware/src/usb/winusb.rs:427]: unfinished comment

**Comment by korken89** [comment]
Very nice!!! I'll give these a try as well with the rusty-probe and the dap-rs extraction I made. Hopefully we can bundle so the descriptors there later and have everything working across multiple MCUs :D

**Comment by mattico** [comment]
I was able to install this on an HS-Probe which was using Zadig-WinUSB, then removed all the existing drivers for its interface, and now it works straight away. Thanks!

File: firmware/src/usb/winusb.rs [hunk 2]
```diff
@@ -1,80 +1,436 @@
...
+
+enum MsDescriptorTypes {
+    Header = 0x0,
+    HeaderConfiguration = 0x1,
+    HeaderFunction = 0x2,
+    CompatibleId = 0x3,
+    RegistryProperty = 0x4,
+}
+
+/// Micorosft OS 2.0 descriptor, according to https://docs.microsoft.com/en-us/windows-hardware/drivers/usbcon/microsoft-os-2-0-descriptors-specification
```
Comment (mattico) [inline firmware/src/usb/winusb.rs:37]: Micorosft > Microsoft

File: firmware/src/usb/winusb.rs [hunk 3]
```diff
@@ -1,80 +1,436 @@
...
+///
+/// For interface ['DAP_V2_INTERFACE'] this configures:
+/// - compatible ID 'WinUSB'
+/// - registry property DeviceInterfaceGUIS = ['{CDB3B5AD-293B-4663-AA36-1AAE46463776}']
```
Comment (mattico) [inline firmware/src/usb/winusb.rs:41]: `DeviceInterfaceGUID`

**Review by Yatekii — APPROVED** [review]
Awesome work :) Thanks a lot!

bors r+

**Pull request merged**